# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1541110107 14400
#      Thu Nov 01 18:08:27 2018 -0400
# Node ID 3fc9d7453ba6d18da787ab7a0a749d0881011a22
# Parent  38a881b8fbec1a9f91330fc2f282f3c7370be471
where possible, use octave_stdout and avoid including <iostream>

* gl-select.cc, gl-select.h, Filter.cpp, History.cpp, load-save.cc,
pr-output.cc, ov-java.cc, oct-parse.yy: Use octave_stdout instead of
std::cout.  Avoid including <iostream> if possible.

diff --git a/libgui/graphics/gl-select.cc b/libgui/graphics/gl-select.cc
--- a/libgui/graphics/gl-select.cc
+++ b/libgui/graphics/gl-select.cc
@@ -19,18 +19,16 @@ along with Octave; see the file COPYING.
 <https://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <iostream>
-
 #include "gl-select.h"
 
 namespace octave
 {
 
   void
   opengl_selector::apply_pick_matrix (void)
   {
diff --git a/libgui/graphics/gl-select.h b/libgui/graphics/gl-select.h
--- a/libgui/graphics/gl-select.h
+++ b/libgui/graphics/gl-select.h
@@ -19,16 +19,17 @@ along with Octave; see the file COPYING.
 <https://www.gnu.org/licenses/>.
 
 */
 
 #if ! defined (octave_gl_select_h)
 #define octave_gl_select_h 1
 
 #include <map>
+#include <string>
 
 #include "gl-render.h"
 #include "oct-opengl.h"
 
 namespace octave
 {
 
   enum select_flags
diff --git a/libgui/qterminal/libqterminal/unix/Filter.cpp b/libgui/qterminal/libqterminal/unix/Filter.cpp
--- a/libgui/qterminal/libqterminal/unix/Filter.cpp
+++ b/libgui/qterminal/libqterminal/unix/Filter.cpp
@@ -18,19 +18,16 @@
     along with this program; if not, write to the Free Software
     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
     02110-1301  USA.
 */
 
 // Own
 #include "unix/Filter.h"
 
-// System
-#include <iostream>
-
 // Qt
 #include <QDesktopServices>
 #include <QAction>
 #include <QApplication>
 #include <QClipboard>
 #include <QtCore/QString>
 
 #include <QtCore/QSharedData>
diff --git a/libgui/qterminal/libqterminal/unix/History.cpp b/libgui/qterminal/libqterminal/unix/History.cpp
--- a/libgui/qterminal/libqterminal/unix/History.cpp
+++ b/libgui/qterminal/libqterminal/unix/History.cpp
@@ -19,17 +19,16 @@
     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
     02110-1301  USA.
 */
 
 // Own
 #include "unix/History.h"
 
 // System
-#include <iostream>
 #include <stdlib.h>
 #include <assert.h>
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/mman.h>
 #include <unistd.h>
 #include <errno.h>
 
diff --git a/libinterp/corefcn/load-save.cc b/libinterp/corefcn/load-save.cc
--- a/libinterp/corefcn/load-save.cc
+++ b/libinterp/corefcn/load-save.cc
@@ -1464,17 +1464,17 @@ namespace octave
 #endif
           // don't insert any commands here!  the brace below must go
           // with the "else" above!
           {
             if (append)
               warning ("save: ignoring -append option for output to stdout");
 
             if (nargout == 0)
-              save_vars (argv, i, argc, std::cout, format,
+              save_vars (argv, i, argc, octave_stdout, format,
                          save_as_floats, true);
             else
               {
                 std::ostringstream output_buf;
                 save_vars (argv, i, argc, output_buf, format,
                            save_as_floats, true);
                 retval = octave_value (output_buf.str());
               }
diff --git a/libinterp/corefcn/pr-output.cc b/libinterp/corefcn/pr-output.cc
--- a/libinterp/corefcn/pr-output.cc
+++ b/libinterp/corefcn/pr-output.cc
@@ -22,17 +22,16 @@ along with Octave; see the file COPYING.
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <cmath>
 
 #include <iomanip>
-#include <iostream>
 #include <limits>
 #include <list>
 #include <sstream>
 #include <string>
 
 #include "Array-util.h"
 #include "CMatrix.h"
 #include "Range.h"
diff --git a/libinterp/octave-value/ov-java.cc b/libinterp/octave-value/ov-java.cc
--- a/libinterp/octave-value/ov-java.cc
+++ b/libinterp/octave-value/ov-java.cc
@@ -30,17 +30,16 @@ along with Octave; see the file COPYING.
 #endif
 
 #if defined (HAVE_WINDOWS_H)
 #  include <windows.h>
 #endif
 
 #include <algorithm>
 #include <fstream>
-#include <iostream>
 #include <map>
 #include <string>
 #include <vector>
 
 #include <clocale>
 
 #include "Cell.h"
 #include "builtin-defun-decls.h"
@@ -256,18 +255,17 @@ namespace octave
 
           while (! js.eof () && ! js.fail ())
             {
               std::getline (js, line);
 
               if (line.find ('-') == 0)
                 java_opts.push_back (line);
               else if (line.length () > 0 && Vdebug_java)
-                std::cerr << "invalid JVM option, skipping: " << line
-                                                              << std::endl;
+                warning ("invalid JVM option, skipping: %s", line.c_str ());
             }
         }
     }
 
   private:
 
     void clean (void)
     {
@@ -292,17 +290,17 @@ namespace octave
           int index = 0;
 
           vm_args.nOptions = java_opts.size ();
           vm_args.options = new JavaVMOption [vm_args.nOptions];
 
           for (const auto& opt : java_opts)
             {
               if (Vdebug_java)
-                std::cout << opt << std::endl;
+                octave_stdout << opt << std::endl;
               vm_args.options[index++].optionString = strsave (opt.c_str ());
             }
 
           java_opts.clear ();
         }
     }
 
   private:
diff --git a/libinterp/parse-tree/oct-parse.yy b/libinterp/parse-tree/oct-parse.yy
--- a/libinterp/parse-tree/oct-parse.yy
+++ b/libinterp/parse-tree/oct-parse.yy
@@ -5051,26 +5051,26 @@ namespace octave
     if (ov_code.is_undefined ())
       return;
 
     if (! ov_code.is_user_code ())
       error ("source: %s is not a script", full_name.c_str ());
 
     if (verbose)
       {
-        std::cout << "executing commands from " << full_name << " ... ";
-        std::cout.flush ();
+        octave_stdout << "executing commands from " << full_name << " ... ";
+        octave_stdout.flush ();
       }
 
     octave_user_code *code = ov_code.user_code_value ();
 
     code->call (tw, 0, octave_value_list ());
 
     if (verbose)
-      std::cout << "done." << std::endl;
+      octave_stdout << "done." << std::endl;
   }
 }
 
 DEFMETHOD (mfilename, interp, args, ,
            doc: /* -*- texinfo -*-
 @deftypefn  {} {} mfilename ()
 @deftypefnx {} {} mfilename ("fullpath")
 @deftypefnx {} {} mfilename ("fullpathext")
